src/bin/vi-draw.rs

```diff
@@ -70,7 +70,7 @@ fn paint_pixels(mut image: *mut u16, padding: u32, width: u32, height: u32, time
             }
             v &= 0x00ffffff;
 
-            unsafe { *image = rgba2yuyv(v, (x & 1) != 0) };
+            unsafe { *image = rgba2yuyv(v as u32, (x & 1) != 0) };
```

Comment:
This as cast can be changed into a try_into

Reply:
There was actually no reason to do this cast, to just after convert it back into `i32` in `rgba2yuyv`, thanks for finding this!

The new version of this commit removed a bunch of useless casts, and made that function const.